corehq/apps/reports/management/commands/project_stats_report.py

```diff
@@ -383,63 +410,45 @@ def _get_table_size(config):
-                        WHERE c.domain = %s
-                        GROUP BY form_id
-                    ) AS d
-                """, [self.domain])
-                result = cursor.fetchall()
-                forms, form_case_updates = (0, 0)
-                if result:
-                    forms, form_case_updates = result[0]
-
-                total_forms += forms
-                total_form_case_updates += form_case_updates
-
-                cursor.execute("""
+        with db_cursor as cursor:
+            cursor.execute("""
+                SELECT COUNT(*) as num_forms, sum(d.count) as num_updates
+                FROM (
                     SELECT COUNT(*) as count
                     FROM form_processor_casetransaction t
                     JOIN form_processor_commcarecasesql c on t.case_id = c.case_id
                     WHERE c.domain = %s
-                """, [self.domain])
-                (transactions,) = cursor.fetchone()
-                total_transactions += transactions
+                    GROUP BY form_id
+                ) AS d
+            """, [self.domain])
+            result = cursor.fetchall()
+            forms, form_case_updates = (0, 0)
+            if result:
+                forms, form_case_updates = result[0]
 
-        resource_model['case_transactions_per_form'] = total_form_case_updates / total_forms
-        resource_model['case_transactions_total'] = total_transactions
+        resource_model['case_transactions_per_form'] = round(form_case_updates / forms, 2)
+        resource_model['case_transactions_total'] = form_case_updates * PARTITIONS_COUNT
 
     def collect_case_indices(self):
-        total_case_indices = 0
-        for db_name in get_db_aliases_for_partitioned_query():
-            db_cursor = connections[db_name].cursor()
+        db_cursor = connections[DB_ALIAS].cursor()
 
-            with db_cursor as cursor:
-                cursor.execute("""
-                    SELECT COUNT(*)
-                    FROM form_processor_commcarecaseindexsql
-                    WHERE domain = %s;
-                """, [self.domain])
-                (case_indices,) = cursor.fetchone()
-                total_case_indices += case_indices
+        with db_cursor as cursor:
+            cursor.execute("""
+                SELECT COUNT(*)
+                FROM form_processor_commcarecaseindexsql
+                WHERE domain = %s;
+            """, [self.domain])
+            (case_indices,) = cursor.fetchone()
 
         total_cases = resource_model['cases_total']
         if total_cases > 0:
-            resource_model['case_index_ratio'] = total_case_indices/total_cases
+            # total_cases are already multiplied by PARTITIONS_COUNT,
+            # so case_indices needs to also be multiplied in order to cancel out that factor
+            resource_model['case_index_ratio'] = (case_indices * PARTITIONS_COUNT) / total_cases
```

Comment:
I was assuming here that the "cases_total" value (which is determined by `CaseES().domain(self.domain).count()`) refers to the _total amount_ of cases, so because the `case_indices` are determined only from 1 shard of the PG DB, I multiplied by PARTITIONS_COUNT in an attempt to estimate the _total amount_ of case indices.